src/main/java/com/fvv/bookstore/view/ProductsReportViewImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.fvv.bookstore.view;
+
+import java.util.List;
+
+import javax.swing.JOptionPane;
+
+import com.fvv.bookstore.bean.Product;
+import com.fvv.bookstore.controller.ProductsReportController;
+import com.fvv.bookstore.controller.ProductsReportControllerImpl;
+import com.fvv.bookstore.exception.ControllerException;
+import com.fvv.bookstore.util.Constants;
+
+/**
+ * ProductsReportView class to view the Products object.
+ * 
+ * @author Fatima Vasquez
+ * <p>Created on 2018</p>
+ * @version 1.0 	
+ *
+ */
+public class ProductsReportViewImpl implements ProductsReportView {
+
+	private final ProductsReportController productsReportController;
+	
+	/**
+	 * Class constructor instantiating a new ProductsReportControllerImpl object.
+	 */
+	public ProductsReportViewImpl() {
+		this.productsReportController = new ProductsReportControllerImpl();
+	}
+	
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void listStockToReplace() {
+		try {
+			StringBuilder sb = new StringBuilder();
+			List<Product> products = this.productsReportController.listStockToReplace();
+			if(products != null && !products.isEmpty()) {
+				for(Product prod : products) {
+					sb.append(prod).append(Constants.LINE_SEPARATOR);
+				}
+				JOptionPane.showMessageDialog(null, sb.toString(), "Listing All Products that Needs to be Replace", 
```

Comment:
you dont need to show the "quantity" here, it obviously will be 0 always

Reply:
The method is using toString from classes bean, and they don't have equal name attributes in each product class, so I can't use get and can't change the toString because is used in other methods